lib/Serialization/Deserialization.cpp

```diff
@@ -183,10 +183,23 @@ void ModuleFile::fatal(llvm::Error error) const {
 SourceLoc ModuleFile::getSourceLoc() const {
   auto &SourceMgr = getContext().Diags.SourceMgr;
   auto filename = getModuleFilename();
-  auto bufferID = SourceMgr.getIDForBufferIdentifier(filename);
-  if (!bufferID)
-    bufferID = SourceMgr.addMemBufferCopy(StringRef(), filename);
-  return SourceMgr.getLocForBufferStart(*bufferID);
+  auto bufferID = SourceMgr.addMemBufferCopy("<binary format>", filename);
+  return SourceMgr.getLocForBufferStart(bufferID);
```

Comment:
Won't this create a new buffer every time (same applies to all the other getSourceLoc's)? Having said that, see my comment below (re. whether we want location at all).

Reply:
Yes. It's far from optimal but I don't think it's critical now since these buffers should only be allocated on failures or when remarks are enabled. We could reuse the buffers but I also want to investigate having diagnostics pointing to a file without a buffer at all, as not showing the `<binary format>` buffer content would be an improvement on its own.

Reply:
If we do get a more optimized version, we could use it for existing services like `extractNearestSourceLoc`.

Reply:
See my other comment. If you really want to keep, fair enough. It just seems like unknown makes more sense in these cases to me.